commit 1e3f9280db69d77b76766cceebb8cc5bb381c1ac
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Mon Feb 13 17:20:08 2017 +0000

    clk-bcm2835: Mark used PLLs and dividers CRITICAL
    
    The VPU configures and relies on several PLLs and dividers. Mark all
    enabled dividers and their PLLs as CRITICAL to prevent the kernel from
    switching them off.
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/drivers/clk/bcm/clk-bcm2835.c b/drivers/clk/bcm/clk-bcm2835.c
index 92fd7cb2cb83..10e6383d7a1f 100644
--- a/drivers/clk/bcm/clk-bcm2835.c
+++ b/drivers/clk/bcm/clk-bcm2835.c
@@ -1367,6 +1367,11 @@ bcm2835_register_pll_divider(struct bcm2835_cprman *cprman,
 	divider->div.hw.init = &init;
 	divider->div.table = NULL;
 
+	if (!(cprman_read(cprman, data->cm_reg) & data->hold_mask)) {
+		init.flags |= CLK_IS_CRITICAL;
+		divider->div.flags |= CLK_IS_CRITICAL;
+	}
+
 	divider->cprman = cprman;
 	divider->data = data;
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1023_1e3f9280db69d77b76766cceebb8cc5bb381c1ac.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1023_1e3f9280db69d77b76766cceebb8cc5bb381c1ac.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
